### ISSUE TITLE
VARCHAR LENGTH additional problem

### ISSUE THREAD

**Issue by Mer0winger** [PomeloFoundation/Pomelo.EntityFrameworkCore.MySql#1323]
Hi, I found this issue in the mentioned subject and try the configuration like this which is working now but before I had same issue.

Following is not working until i changed the `varchar` to `varchar(15)` but as mentioned in #1050 it should also work only with the `.HasMaxLength(15)` at the end! 

`builder.Property(x => x.nationalidnumber).HasColumnName(@"NationalIDNumber").HasColumnType(@"**varchar**") .IsRequired().ValueGeneratedNever().HasMaxLength(15);`

In the answer of  #1050 there was a sample code but there the same workaround was done to put the length into `HasColumnType`. Same happen also for `varbinary` maybe also other types I haven't tested all of them. 

Working if I add all the lengths to the `HasColumnType` as below:

```C#
public void Configure(EntityTypeBuilder<employeeEntity> builder)
{
    builder.ToTable(@"employee", @"adventureworks");
    builder.Property(x => x.employeeid).HasColumnName(@"EmployeeID").HasColumnType(@"int").IsRequired().ValueGeneratedNever();
    builder.Property(x => x.nationalidnumber).HasColumnName(@"NationalIDNumber").HasColumnType(@"**varchar(15)**").IsRequired().ValueGeneratedNever().HasMaxLength(15);
    builder.Property(x => x.contactid).HasColumnName(@"ContactID").HasColumnType(@"int").IsRequired().ValueGeneratedNever();
    builder.Property(x => x.loginid).HasColumnName(@"LoginID").HasColumnType(@"varchar(256)").IsRequired().ValueGeneratedNever().HasMaxLength(256);
    builder.Property(x => x.managerid).HasColumnName(@"ManagerID").HasColumnType(@"int").ValueGeneratedNever();
    builder.Property(x => x.title).HasColumnName(@"Title").HasColumnType(@"varchar(50)").IsRequired().ValueGeneratedNever().HasMaxLength(50);
    builder.Property(x => x.birthdate).HasColumnName(@"BirthDate").HasColumnType(@"datetime").IsRequired().ValueGeneratedNever();
    builder.Property(x => x.maritalstatus).HasColumnName(@"MaritalStatus").HasColumnType(@"varchar(1)").IsRequired().ValueGeneratedNever().HasMaxLength(1);
    builder.Property(x => x.gender).HasColumnName(@"Gender").HasColumnType(@"varchar(1)").IsRequired().ValueGeneratedNever().HasMaxLength(1);
    builder.Property(x => x.hiredate).HasColumnName(@"HireDate").HasColumnType(@"datetime").IsRequired().ValueGeneratedNever();
    builder.Property(x => x.salariedflag).HasColumnName(@"SalariedFlag").HasColumnType(@"bit").IsRequired().ValueGeneratedNever();
    builder.Property(x => x.vacationhours).HasColumnName(@"VacationHours").HasColumnType(@"smallint").IsRequired().ValueGeneratedNever();
    builder.Property(x => x.sickleavehours).HasColumnName(@"SickLeaveHours").HasColumnType(@"smallint").IsRequired().ValueGeneratedNever();
    builder.Property(x => x.currentflag).HasColumnName(@"CurrentFlag").HasColumnType(@"bit").IsRequired().ValueGeneratedNever();
    builder.Property(x => x.rowguid).HasColumnName(@"rowguid").HasColumnType(@"varbinary(16)").IsRequired().ValueGeneratedNever().HasMaxLength(16);
    builder.Property(x => x.modifieddate).HasColumnName(@"ModifiedDate").HasColumnType(@"timestamp").IsRequired().ValueGeneratedOnAdd().HasDefaultValueSql(@"CURRENT_TIMESTAMP");
    builder.HasKey(@"employeeid");
    builder.HasMany(x => x.employeeEntities_managerid).WithOne(op => op.employeeEntity_managerid).OnDelete(DeleteBehavior.Restrict).IsRequired(false).HasForeignKey(@"managerid");
    builder.HasOne(x => x.employeeEntity_managerid).WithMany(op => op.employeeEntities_managerid).OnDelete(DeleteBehavior.Restrict).IsRequired(false).HasForeignKey(@"managerid");

    CustomizeConfiguration(builder);
}
```

**Comment by mguinness** [comment]
Could be related to upstream issue https://github.com/dotnet/efcore/issues/15159.

**Comment by lauxjpn** [comment]
@Mer0winger You did not fill out our issue template. What version of Pomelo and EF Core are you using?

@mguinness Yes, this has indeed been fixed in EF Core for 5.0.0, but we haven't adopted it yet.

Currently, with `Name` being a `string` property:

```c#
// This works:
entity.Property(e => e.Name)
    .HasColumnType("varchar(200)");

// This also works:
entity.Property(e => e.Name)
    .HasMaxLength(200);

// This does not work:
entity.Property(e => e.Name)
    .HasColumnType("varchar")
    .HasMaxLength(200);
```

Since EF Core has fixed it upstream, we should make this scenario work for 5.0.0 as well.
